presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java

```diff
@@ -114,6 +114,7 @@ public void testLargeData()
         assertQuery("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name LIKE 'test_t_ble1'", "VALUES 2");
         assertQuery("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name LIKE 'test_t_ble1' AND table_name IN ('test_table1', 'test_table2')", "VALUES 2");
         assertQuery("SELECT count(*) from test_catalog.information_schema.columns WHERE table_schema = 'test_schema1' AND table_name = 'test_table1'", "VALUES 100");
+        assertQuery("SELECT count(*) from test_catalog.information_schema.columns WHERE table_catalog = 'wrong'", "VALUES 0");
```

Comment:
Is this correctness fix or speedup (short-circuiting)?

Reply:
speedup (short-circuiting)

Reply:
(in the latter case, it'd be good to have a test that counts metadata calls)

Reply:
> (in the latter case, it'd be good to have a test that counts metadata calls)

There is such, notice that count of metadata calls didn't change.